code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -90,6 +91,28 @@
 
 	return dat
 
+/mob/living/simple_animal/bot/floorbot/attackby(obj/item/W , mob/user, params)
+	if(istype(W, /obj/item/stack/tile/plasteel))
+		user << "<span class='notice'>The floorbot can produce normal tiles itself.</span>"
+		return
+	if(specialtiles != 0 && istype(W, /obj/item/stack/tile/))
+		var/obj/item/stack/tile/usedtile = W
+		if(usedtile != tiletype)
+			user << "<span class='warning'>Different custom tiles are already inside the floorbot.</span>"
+			return
+	if(istype(W, /obj/item/stack/tile/))
+		tiletype = W
```

Comment:
You've set a reference, which is a permanent link to that object (so long as it exists), you could force the user to drop the item and put the item INSIDE the bot, which means the object only gets destroyed if the bot does, meaning that the bot always has safe access to it.